### ISSUE TITLE
Show result of admin_post_thumbnail_html filter in Featured Image panel

### ISSUE THREAD

**Issue by jaymanpandya** [WordPress/gutenberg#12742]
**Right now users do not understand where and how the uploaded Featured Image will be shown on the site**

The area from where Featured Image is uploaded is very concise and does not have information on how the uploaded Featured Image will be used in the theme. This can cause confusion amongst the users who have not made their own theme. This issue can amplify if the uploaded image is used in different size through out the site and users will not test across the site and won't know how the image is appearing. 

We can have short descriptions as suggested in the examples below to describe the behaviour of the uploaded Featured Image on the site.

**Examples:**
- The featured image will be shown on archives next to the optional excerpt of posts.
- 3 different sizes of featured image will be used: 50px - shown next to the post title in sidebar headlines. 150px - shown next to post excerpts on the homepage and archives. 500px - shown in the slider on the homepage.

This ticket was created based on a very old ticket on Trac ([https://core.trac.wordpress.org/ticket/13522](https://core.trac.wordpress.org/ticket/13522))

**Comment by joyously** [comment]
Can we reinstate the filter that used to exist for this in the old editor? `admin_post_thumbnail_html`

**Comment by jerclarke** [comment]
That would be a good start of course. I (original ticket author) didn't realize Gutenberg had stomped that filter. 

Definitely need that filter to continue working, as it's the only reason this ticket was allowed to get so stale. 

The point of the ticket was to create an easy-to-use API to register this information for display, like the description field that widgets and sidebars have.

**Comment by melchoyce** [comment]
@joyously Can you make a new issue for that?

**Comment by joyously** [comment]
No, as it's pretty much the conclusion from the Trac ticket discussion, so it's a big part of this issue.

**Comment by mapk** [comment]
The Featured Image module in the Inspector is void of text, so it does require the user to have some sort of knowledge to interact successfully with it. Because of this reason, I'm for adding _something_ to the module to help explain what a Feature Image is, or how it's used.

**Current Featured Image module**

![Screen Shot 2019-06-04 at 10 27 05 AM](https://user-images.githubusercontent.com/617986/58907344-cb0e4c00-86c2-11e9-9a63-4bc9358b6f06.png)

**Comment by mapk** [comment]
In today's Design Team slack chat, we agreed that it would be great to get [this filter implemented back](https://github.com/WordPress/gutenberg/issues/12742#issuecomment-467187842). This would allow the themes to provide a Featured Image description.

**Comment by quasiDigi** [comment]
It would be nice to have the `admin_post_thumbnail_html` back. So we can add extra text and also extra fields. The best example is the "_show feature image on top of the post page_" option. 
Thx!

**Comment by rickyfang** [comment]
Any updates on this feature guys? Our users would really appreciate it.

**Comment by skorasaurus** [comment]
Does the editor.PostFeaturedImage filter fulfill what you're looking to do? 

The [PR implementing this filter](https://github.com/WordPress/gutenberg/pull/6940) mentions that it acts like admin_post_thumbnail_html. More info at https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/post-featured-image/README.md

**Comment by joyously** [comment]
If that filter simply "acts like" a PHP filter, it's not the same thing as being a PHP filter. What should happen is the PHP filter that is [documented](https://developer.wordpress.org/reference/hooks/admin_post_thumbnail_html/) and called by existing code should affect the new editor the same way it affects the old editor, without needing JS code to be added by the legacy plugin or theme in order to work.

**Comment by eric-michel** [comment]
@joyously, agree here. I was just trying to use this documented filter before realizing it didn't work in Gutenberg. Our theme is set up to use Featured Images differently depending on settlings chosen in the customizer, so we need a PHP filter that can conditionally change the help text based on those selected settings.

The Featured Image input is incredibly unclear on what it does, and is indeed confusing our clients. Is there any inertia to get this fixed?

**Comment by joyously** [comment]
It seems like it would be pretty simple. Maybe if you ask in the #core-editor channel for someone to add this to the next milestone?
_By the way, you need momentum to overcome inertia. There's plenty of inertia already._

**Comment by eric-michel** [comment]
@joyously Thanks! I will see about popping in there to get a feel for where they are. I wish I could work on this myself but am still way out of my depth with React.

**Comment by noisysocks** [comment]
Closing this in favour of https://github.com/WordPress/gutenberg/issues/30904.

**Comment by joyously** [comment]
The other issue (30904) doesn't even mention anything about the PHP filter.

**Comment by noisysocks** [comment]
My mistake. Let's keep https://github.com/WordPress/gutenberg/issues/30904 for adding help text and this issue for respecting the existing Core filter.

**Comment by Mamaduka** [comment]
I believe the path forward is to use the `editor.PostFeaturedImage` filter.

Here's an example migration guide - https://github.com/danielbachhuber/gutenberg-migration-guide/blob/master/filter-admin-post-thumbnail-html.md.

I think we can close this as "won't fix" and "has migration path."

**Comment by joyously** [comment]
Why choose to make every theme or plugin add JS, when it would be better for core to do the migration instead? Why remove functionality that has been there for a long time and is still [documented](https://developer.wordpress.org/reference/hooks/admin_post_thumbnail_html/)?
Please reread the comments on this issue.

**Comment by benwallis** [comment]
I think this filter `admin_post_thumbnail_html` is broken in the Classic Editor now. When I use it, the Featured Image assigned to a post cannot be updated/saved/deleted.

**Comment by joyously** [comment]
> I think this filter `admin_post_thumbnail_html` is broken in the Classic Editor now. When I use it, the Featured Image assigned to a post cannot be updated/saved/deleted.

Because of this comment, I tested it on WP 6.1. That filter works fine in Classic editor, as does the setting and deleting the featured image.
This issue is about having that filter be useful in the block editor.

**Comment by benwallis** [comment]
> > I think this filter `admin_post_thumbnail_html` is broken in the Classic Editor now. When I use it, the Featured Image assigned to a post cannot be updated/saved/deleted.
> 
> Because of this comment, I tested it on WP 6.1. That filter works fine in Classic editor, as does the setting and deleting the featured image. This issue is about having that filter be useful in the block editor.

Ah, sorry. I was checking $screen-post_type in the filter but apparently the ajax call also applies the filter (where it is not set), which was causing a PHP warning.

**Comment by BlackStar1991** [comment]
Has anyone found an answer to this question in 2023? WP vercion 6.2

**Comment by Mamaduka** [comment]
Closing this as "won't fix". The old PHP filters can't be used with React; the editor already provides `editor.PostFeaturedImage`, which lets you override the Featured Image output in the block editor.

The document functionality for `admin_post_thumbnail_html` still works for the Classic Editor.
